### PR TITLE
Update bulletproof-gadgets version

### PR DESCRIPTION
**Pull request by filiplazovic**

### CODE REVIEW

**Comment by filiplazovic** [comment]
Noticed some comments are missing after some rebase in the past, gotta fix it here

File: pallets/merkle/src/lib.rs [hunk 1]
```diff
@@ -752,14 +755,14 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
-	pub fn add_leaf(tree: &mut GroupTree, data: ScalarData, h: &Poseidon) {
+	pub fn add_leaf(tree: &mut GroupTree, data: ScalarData, zero_tee: &Vec<[u8; 32]>, h: &Poseidon) {
```
Comment (shekohex) [inline pallets/merkle/src/lib.rs:758]:
```suggestion
	pub fn add_leaf(tree: &mut GroupTree, data: ScalarData, zero_tree: &Vec<[u8; 32]>, h: &Poseidon) {
```

File: pallets/merkle/src/lib.rs [hunk 2]
```diff
@@ -752,14 +755,14 @@ impl<T: Config> Pallet<T> {
...
 		let mut edge_index = tree.leaf_count;
 		let mut hash = data.0;
 		// Update the tree
 		for i in 0..tree.edge_nodes.len() {
 			hash = if edge_index % 2 == 0 {
 				tree.edge_nodes[i] = ScalarData(hash);
-				let zero_h = Scalar::from_bytes_mod_order(ZERO_TREE[i]);
+				let zero_h = Scalar::from_bytes_mod_order(zero_tee[i]);
```
Comment (shekohex) [inline pallets/merkle/src/lib.rs:765]:
```suggestion
				let zero_h = Scalar::from_bytes_mod_order(zero_tree[i]);
```

File: pallets/merkle/src/traits.rs [hunk 3]
```diff
@@ -5,23 +5,38 @@ use bulletproofs::PedersenGens;
 pub use frame_support::dispatch;
 use sp_std::vec::Vec;
 
+/// Group trait definition to be used in other pallets
 pub trait Group<AccountId, BlockNumber, GroupId> {
+	/// Check if nullifier is already used, in which case throws an error
```
Comment (shekohex) [inline pallets/merkle/src/traits.rs:10]:
```suggestion
	/// Check if nullifier is already used, in which case return an error
```

**Review by shekohex — CHANGES_REQUESTED** [review]
Sounds Good!
Just a few changes, I made it as a suggestion so you can do it quickly from here!

**Review by shekohex — APPROVED** [review]

**Pull request merged**